uhu/core/objects.py

```diff
@@ -52,10 +52,16 @@ def load(self, callback=None):
             obj.load(callback=callback)
         call(callback, 'finish_objects_load')
 
+    def _check_duplicate_object_entry(self, entry):
+        for obj in self.objects:
+            if(obj.__eq__(entry)):
+                raise DuplicateObjectEntryError("Object duplicate rejected. Aborting...")
```

Comment:
Couldn't this to be done similar to:
```
   if entry in self.objects:
       raise ...
```

Reply:
I thought it was possible. It turns out self.objects is a list of tuples and entry, a list of objects. Even casting entry to a tuple of objects didn't do the trick, in a way that what you suggested is possible. The way I found was through dictionary comparison (__eq__ method); what is weird is that "in" or "==" use this method (__eq__) under the hood.